pkg/cmd/bind.go

```diff
@@ -206,6 +193,34 @@ func (o *bindCmdOptions) run(args []string) error {
 	return nil
 }
 
+func showOutput(cmd *cobra.Command, binding *v1alpha1.KameletBinding, outputFormat string) error {
```

Comment:
The _schema_ is the concept that is responsible to map GKV and types, so it may be used here, rather than doing it ourselves. It implements the `runtime.ObjectTyper` interface, that can be used to add the GKV information. Also it may be possible to rely on the `cli-runtime` package, which is used by `kubectl`:

https://github.com/kubernetes/cli-runtime/blob/c2f4736118ac4e1fd42fdd347954747433ca4fb7/pkg/printers/typesetter.go#L35

Reply:
I've tried to follow the approach you proposed. In order to make it works, we need to use a newer version of Golang (1.16) as those libraries seems to require it. [From a separate discussion](https://github.com/apache/camel-k/issues/2579) I understood we're not yet ready for it. Should we progress and then apply your suggestion once we move to 1.16?

Reply:
Doesn't it work by depending on the 1.20 version: https://github.com/kubernetes/cli-runtime/blob/release-1.20/go.mod?

Otherwise, following the Golang version from k8s is rather a rule of thumb, and more a minorant. We can try upgrading to 1.16 if needs be.

Reply:
Using `k8s.io/cli-runtime v0.20.2` seems okey with go 1.5. I'm trying to see how I can use it accordingly now, thanks.

Reply:
I've just committed the changes. Thanks for the suggestion, it looks very neat now and hopefully we can reuse this logic for any other runtime object! Please, have a look if it fits with what you had in mind.

Reply:
Great, my understanding is that the scheme abstraction is meant to be used so that it works with any runtime object that's registered into the scheme:

https://github.com/kubernetes/kubectl/blob/4defba0cec1f594eb410c69bff05b51cddfba8ff/pkg/cmd/get/get.go#L266

Every online command should already have the scheme initialised correctly, and accessible from `client.GetScheme()`.